Parser/Node.cs

```diff
@@ -20,7 +20,23 @@ public record NBlock (NDeclarations Decls, NCompoundStmt Body) : Node {
 }
 
 // The declarations section precedes the body of every block
-public record NDeclarations (NVarDecl[] Vars) : Node {
+public record NDeclarations (NVarDecl[] Vars, NProcFnDecls ProcFns) : Node {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+public record NProcFnDecls (NProcDecl[] Procs, NFnDecl[] Fns) : Node {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+public record NProcDecl (Token Name, NParams Params, NBlock Block) : Node {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+public record NFnDecl (Token Name, NParams Params, NType Type, NBlock Block) : Node {

```

Comment:
Can you merge NProcDecl and NFnDecl into a single record (NFnDecl). By setting the return type to NType.Void we can indicate that this is a procedure.